src/Files.App/Commands/Manager/CommandManager.cs

```diff
@@ -310,75 +353,135 @@ private class NoneCommand : IRichCommand
 
-			public bool IsToggle => action is IToggleAction;
+			public string? HotKeyText
+			{
+				get
+				{
+					string text = HotKeys.Label;
+					if (string.IsNullOrEmpty(text))
+						return null;
+					return text;
+				}
+			}
+
+			private HotKeyCollection hotKeys;
+			public HotKeyCollection HotKeys
+			{
+				get => hotKeys;
+				set
+				{
+					if (hotKeys == value)
+						return;
+
+					string code = Code.ToString();
+					var customs = new Dictionary<string, string>(manager.settings.Actions);
+
+					if (!customs.ContainsKey(code))
+						customs.Add(code, value.Code);
+					else if (value != GetHotKeys(Action))
+						customs[code] = value.Code;
+					else
+						customs.Remove(code);
+
+					manager.settings.Actions = customs;
+				}
+			}
+
+			public bool IsToggle => Action is IToggleAction;
 
 			public bool IsOn
 			{
-				get => action is IToggleAction toggleAction && toggleAction.IsOn;
+				get => Action is IToggleAction toggleAction && toggleAction.IsOn;
 				set
 				{
-					if (action is IToggleAction toggleAction && toggleAction.IsOn != value)
+					if (Action is IToggleAction toggleAction && toggleAction.IsOn != value)
 						Execute(null);
 				}
 			}
 
-			public bool IsExecutable => action.IsExecutable;
+			public bool IsExecutable => Action.IsExecutable;
 
-			public ActionCommand(CommandCodes code, IAction action)
+			public ActionCommand(CommandManager manager, CommandCodes code, IAction action)
 			{
+				this.manager = manager;
 				Code = code;
-				this.action = action;
+				Action = action;
 				Icon = action.Glyph.ToIcon();
 				FontIcon = action.Glyph.ToFontIcon();
 				OpacityStyle = action.Glyph.ToOpacityStyle();
-				HotKeyText = GetHotKeyText();
-				LabelWithHotKey = HotKeyText is null ? Label : $"{Label} ({HotKeyText})";
+				hotKeys = GetHotKeys(action);
 
 				if (action is INotifyPropertyChanging notifyPropertyChanging)
 					notifyPropertyChanging.PropertyChanging += Action_PropertyChanging;
 				if (action is INotifyPropertyChanged notifyPropertyChanged)
 					notifyPropertyChanged.PropertyChanged += Action_PropertyChanged;
 			}
 
-			public bool CanExecute(object? parameter) => action.IsExecutable;
+			public bool CanExecute(object? parameter) => Action.IsExecutable;
 			public async void Execute(object? parameter) => await ExecuteAsync();
 
 			public async Task ExecuteAsync()
 			{
 				if (IsExecutable)
-					await action.ExecuteAsync();
+					await Action.ExecuteAsync();
 			}
 
 			public async void ExecuteTapped(object sender, TappedRoutedEventArgs e) => await ExecuteAsync();
 
+			public void ResetHotKeys()
+			{
+				if (!IsCustomHotKeys)
+					return;
+
+				var customs = new Dictionary<string, string>(manager.settings.Actions);
+				customs.Remove(Code.ToString());
+				manager.settings.Actions = customs;
+			}
+
+			internal void UpdateHotKeys(bool isCustom, HotKeyCollection hotKeys)
+			{
+				if (Code is CommandCodes.SelectAll or CommandCodes.ClearSelection)
+				{
+
+				}
```

Comment:
What is the use of that empty condition?

Reply:
Good catch!

Reply:
Sorry. It's fixed.